src/indexer.cc

```diff
@@ -830,6 +831,125 @@ class IndexDataConsumer : public index::IndexDataConsumer {
            D->getKind() == Decl::CXXConversion))
         role = Role(role | Role::Implicit);
       do_def_decl(func);
+
+      if (is_def) {
+        class FuncVisitor : public RecursiveASTVisitor<FuncVisitor> {
+          public:
+            FuncVisitor(IndexDataConsumer* parent, 
+    IndexFile *db, SourceManager& SM, const LangOptions &Lang, int lid, const Decl* D) : parent(parent), db(db), SM(SM), Lang(Lang), lid(lid), root(D) {} 
+            bool VisitFunctionDecl(FunctionDecl* decl) {
+              return true;
+            }
+            bool VisitBinAssign(BinaryOperator* binop) {
+              AddDataFlow(binop->getLHS(), binop->getRHS());
+              return RecursiveASTVisitor::VisitBinAssign(binop);
+            }
+            bool VisitVarDecl(VarDecl* decl) {
+              if (decl->hasInit()) {
```

Comment:
These parentheses wrapping simple statements are very annoying. LLVM style tends to avoid them.